src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -78,7 +161,13 @@ public void OnEndRequest(HttpContext httpContext, long timestamp)
         public void OnHostingException(HttpContext httpContext, Exception exception)
         {
             this.OnException(httpContext, exception);
-            this.EndRequest(httpContext, Stopwatch.GetTimestamp());
+
+            // In AspNetCore 1.0, when an exception is unhandled, it will only send UnhandledException event but not EndRequest event, so we need to EndRequest at here.
```

Comment:
nitpick: "...when an exception is unhandled it will only send the UnhandledException event, but not the EndRequest event, so we need to call EndRequest here."